java-checks/src/main/java/org/sonar/java/checks/PreferStreamAnyMatchCheck.java

```diff
@@ -60,20 +68,60 @@
 
   @Override
   protected List<MethodMatcher> getMethodInvocationMatchers() {
-    return Stream.of("java.util.Optional", "java.util.OptionalInt", "java.util.OptionalLong", "java.util.OptionalDouble")
+    List<MethodMatcher> matchers = new ArrayList<>();
+    Stream.of("java.util.Optional", "java.util.OptionalInt", "java.util.OptionalLong", "java.util.OptionalDouble")
       .map(type -> MethodMatcher.create().typeDefinition(type).name("isPresent").withoutParameter())
-      .collect(Collectors.toList());
+      .forEach(matchers::add);
+    STREAM_TYPES.stream()
+      .map(type -> MethodMatcher.create().typeDefinition(type).name("anyMatch").addParameter("java.util.function.Predicate"))
+      .forEach(matchers::add);
+    return matchers;
   }
 
   @Override
-  protected void onMethodInvocationFound(MethodInvocationTree isPresentMIT) {
+  protected void onMethodInvocationFound(MethodInvocationTree mit) {
+    if (mit.symbol().name().equals("isPresent")) {
+      handleIsPresent(mit);
+    } else if (mit.symbol().name().equals("anyMatch")) {
+      handleAnyMatch(mit);
+    }
+  }
+
+  private void handleAnyMatch(MethodInvocationTree anyMatchMIT) {
+    ExpressionTree predicate = anyMatchMIT.arguments().get(0);
+    if (anyMatchMIT.parent().is(Tree.Kind.LOGICAL_COMPLEMENT)) {
+      if (predicate.is(Tree.Kind.LAMBDA_EXPRESSION) && ((LambdaExpressionTree) predicate).body().is(Tree.Kind.LOGICAL_COMPLEMENT)) {
+        // !stream.anyMatch(x -> !(...))
+        context.reportIssue(this, MethodsHelper.methodName(anyMatchMIT),
```

Comment:
I think `MethodsHelper.methodName(anyMatchMIT)` could be extracted in a dedicated `reportTree`, as it could be used in 3 different places.